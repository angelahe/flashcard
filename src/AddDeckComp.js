import React from "react"

class AddDeckComp extends React.Component {
    constructor() {
        super()
        this.state = {
            currentDeck : null
        }
    }

    handleAddClick = () => {
        console.log("in handleAddClick")
        //call 
        const deckId = "FakeID"
        this.setState ({currentDeck : deckId});
    } 

    render() {
        return (
            <div>
                <p>Welcome to Flashcards</p>
                <button onClick={this.handleAddClick}>Add Deck</button>
                {this.state.currentDeck ? <p>current deck: {this.state.currentDeck}</p> : null}
            </div>
        )
    }
}
export default AddDeckComp
