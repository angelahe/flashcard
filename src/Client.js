// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(L1word, L2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  return fetch(`api/card?EN_word=${L1word}&ES_word=${L2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
