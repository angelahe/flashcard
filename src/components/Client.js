// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(deck => deck.id);
}

function createCard(deckId, l1word, l2word, imageUrl, imageId) {
  return fetch(`api/deck/${deckId}/card?L1_word=${l1word}&L2_word=${l2word}
    &img_url=${imageUrl}&img_id=${imageId}`,
  {
    method: 'POST',
  })
    .then(response => response.json())
    .then(card => card.card_id);
}

function getDecks() {
  return fetch('api/deck/all', {
    method: 'GET',
  })
    .then(response => response.json());
}

function getCards(deckId) {
  return fetch(`api/deck/${deckId}/cards`, {
    method: 'GET',
  })
    .then(response => response.json());
}

const Client = {
  createDeck, createCard, getDecks, getCards,
};
export default Client;
