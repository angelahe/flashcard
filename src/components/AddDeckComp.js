import React from 'react';
import PropTypes from 'prop-types';
import Client from './Client';
import done from '../img/buttons/done_FFFFFF.png';

class AddDeckComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentDeck: '',
      nameValue: '',
      keyValue: '',
      orderValue: 0,
    };
  }

  handleNameChange = (event) => {
    this.setState({ nameValue: event.target.value });
  };

  handleKeyChange = (event) => {
    this.setState({ keyValue: event.target.value });
  };

  handleOrderChange = (event) => {
    this.setState({ orderValue: event.target.value });
  };

  handleAddDeckClick = () => {
    const { onDeckAdded } = this.props;

    Client.createDeck()
      .then(id => this.setState({ currentDeck: id },
        // eslint-disable-next-line react/destructuring-assignment
        () => onDeckAdded(this.state.currentDeck)));
  }

  render() {
    const {
      currentDeck,
      nameValue,
      keyValue,
      orderValue,
    } = this.state;
    return (
      <div className="AddDeckComp">
        <div>
          {currentDeck !== ''
            ? <p>current deck: {currentDeck}</p>
            : null
          }
        </div>

        <div>
          <br /> <br />
          <div className="LineContainer">
            <span className="DetailText">Name:</span>
            <input className="DeckName DetailText" value={nameValue} onChange={this.handleNameChange} />
            <br />
          </div>
          <br />
          <div className="LineContainer">
            <span className="DetailText">Key:</span>
            <input className="KeyValue DetailText" value={keyValue} onChange={this.handleKeyChange} />
          </div>
          <br />
          <div className="LineContainer">
            <span className="DetailText">Order:</span>
            <input className="DeckOrder DetailText" type="Number" value={orderValue} onChange={this.handleOrderChange} />
          </div>
          <br />
          <button className="AppBtn AddDeckButton" type="button" onClick={this.handleAddDeckClick}>
            <img className="btnImg" src={done} alt="Done" />
          </button>
        </div>
      </div>
    );
  }
}

AddDeckComp.propTypes = {
  onDeckAdded: PropTypes.func.isRequired,
};
export default AddDeckComp;
