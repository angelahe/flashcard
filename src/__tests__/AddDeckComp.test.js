import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import AddDeckComp from '../AddDeckComp';
import Client from '../Client';

jest.mock('../Client');

it('renders without crashing', () => {
  shallow(<AddDeckComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddDeckComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('does not show the current deck when there is no deck', () => {
  const component = shallow(<AddDeckComp />);
  expect(component.html()).not.toContain('current deck');
});

it('shows the current deck when there is one', () => {
  const deckId = '12345-ABCD';
  const component = shallow(<AddDeckComp />);
  component.setState({ currentDeck: deckId });
  component.update();

  expect(component.html()).toContain(`current deck: ${deckId}`);
});

it('should set the state when createDeck completes', (done) => {
  const deckId = '12345-ABCD';
  Client.createDeck = jest.fn(() => Promise.resolve(deckId));

  const component = shallow(<AddDeckComp />);
  const button = component.find('button').first();
  button.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentDeck).toBe(deckId);
    done();
  });
});
