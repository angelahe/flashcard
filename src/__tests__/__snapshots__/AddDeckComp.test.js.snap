// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDeckComp renders correctly 1`] = `
<div>
  <div />
  <div>
    <br />
     
    <br />
    <div
      className="LineContainer"
    >
      <span
        className="DetailText"
      >
        Name:
      </span>
      <input
        className="DetailText"
        onChange={[Function]}
        value=""
      />
      <br />
    </div>
    <br />
    <div
      className="LineContainer"
    >
      <span
        className="DetailText"
      >
        Key:
      </span>
      <input
        className="DetailText"
        onChange={[Function]}
        value=""
      />
    </div>
    <br />
    <div
      className="LineContainer"
    >
      <span
        className="DetailText"
      >
        Order:
      </span>
      <input
        className="DetailText"
        onChange={[Function]}
        type="Number"
        value={0}
      />
    </div>
    <br />
    <button
      className="AppBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Done"
        className="btnImg"
        src="done_FFFFFF.png"
      />
    </button>
  </div>
</div>
`;
